### PR TITLE
Feature/basic auth

### PR DESCRIPTION
**Pull request by lstuma**

### CODE REVIEW

**Comment by lstuma** [comment]
waiting for review from @raspitim

**Comment by lstuma** [comment]
ok gotta fix these errors first

**Review by raspitim — APPROVED** [review]

**Review by raspitim — CHANGES_REQUESTED** [review]
test fails

**Review by raspitim — APPROVED** [review]
Nice wrapping 🌯🌯🌯🌯

**Pull request merged**
